client/signup/steps/import-from/medium/index.tsx

```diff
@@ -1,39 +1,125 @@
-import { Hooray, SubTitle, Title } from '@automattic/onboarding';
+import { Hooray, Progress, SubTitle, Title } from '@automattic/onboarding';
 import { useI18n } from '@wordpress/react-i18n';
 import classnames from 'classnames';
-import React from 'react';
+import { translate, TranslateOptions, TranslateOptionsText } from 'i18n-calypso';
+import React, { useEffect } from 'react';
 import { connect } from 'react-redux';
+import InlineSupportLink from 'calypso/components/inline-support-link';
+import { ProgressBar } from 'calypso/devdocs/design/playground-scope';
+import importerConfig from 'calypso/lib/importer/importer-config';
+import { calculateProgress } from 'calypso/my-sites/importer/importing-pane';
 import { resetImport, startImport } from 'calypso/state/imports/actions';
+import { appStates } from 'calypso/state/imports/constants';
 import { importSite } from 'calypso/state/imports/site-importer/actions';
 import DoneButton from '../components/done-button';
+import GettingStartedVideo from '../components/getting-started-video';
+import ImporterDrag, { Site } from '../components/importer-drag';
 import { Importer, ImportJob, ImportJobParams } from '../types';
-
+import { getImporterTypeForEngine } from '../util';
 import './style.scss';
 
 interface Props {
 	job?: ImportJob;
 	run: boolean;
 	siteId: number;
+	site: Site;
 	siteSlug: string;
 	fromSite: string;
-	importSite?: ( params: ImportJobParams ) => void;
-	startImport?: ( siteId: number, type: string ) => void;
-	resetImport?: ( siteId: number, importerId: string ) => void;
+	importSite: ( params: ImportJobParams ) => void;
+	startImport: ( siteId: number, type: string ) => void;
+	resetImport: ( siteId: number, importerId: string ) => void;
 }
 export const MediumImporter: React.FunctionComponent< Props > = ( props ) => {
 	const importer: Importer = 'medium';
 	const { __ } = useI18n();
-	const { job, siteId, siteSlug } = props;
+	const { job, siteId, site, siteSlug, fromSite, importSite, startImport, resetImport } = props;
+	const importerData = importerConfig().medium;
+
+	populateMessages();
+
+	/**
+	 * Effects
+	 */
+	useEffect( runImport, [ job ] );
```

Comment:
In the meantime, I've improved the way of deciding when to run the import. Maybe you can reuse it here.
https://github.com/Automattic/wp-calypso/blob/trunk/client/signup/steps/import-from/wix/index.tsx#L37-L74